pkg/util/preconditions.go

```diff
@@ -0,0 +1,49 @@
+package util
+
+import (
+	"context"
+	"os/exec"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+)
+
+func ValidateEnvironment() error {
+	_, err := exec.LookPath("docker")
+	if err != nil {
+		return errors.Wrap(err, "could not find docker in path")
+	}
+
+	_, err = exec.LookPath("aws")
+	if err != nil {
+		return errors.Wrap(err, "could not find aws cli in path")
+	}
+
+	_, err = exec.LookPath("terraform")
+	if err != nil {
+		return errors.Wrap(err, "could not find terraform in path")
+	}
+
+	client, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return errors.Wrap(err, "docker engine is not running")
+	}
+
+	_, err = client.ContainerList(context.Background(), types.ContainerListOptions{})
+	return errors.Wrap(err, "cannot connect to docker engine")
```

Comment:
I wonder if we should "collect" all these errors and return them as a combined error -- see `multierror` in other parts of the repo. That way it is easy for users to see all the things they should fix.

also, how about adding the ssm session manager check as well?